.github/workflows/FormatCheck.yml

```diff
@@ -0,0 +1,44 @@
+name: Format Check
+on:
+  push:
+    branches:
+      - 'main'
+    tags: 
+      - '*'
+  pull_request:
+    branches:
+      - 'main'
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: julia-actions/setup-julia@v2
```

Comment:
Can we use the cache action, or not?

Reply:
Otherwise we can merge.

Reply:
not sure whether need the cache, since this CI only need to install `JuliaFormatter.jl`